File: nuido_flow_data/flows/data/lookup_node.py
# THIS FILE IS A PART OF PUBLIC REPOSITORY https://github.com/yonitjio/exploring-odoo
# 
# This software is released under the MIT License.
# https://opensource.org/licenses/MIT
# 
# THIS SOFTWARE IS EXPERIMENTAL AND FOR EDUCATIONAL PURPOSE ONLY.
# DO NOT USE IT IN PRODUCTION.

import logging

_logger = logging.getLogger(__name__)

from odoo.addons.nuido_flow.flows.core.base_node import BaseNode

class LookupNode(BaseNode):
    def process(self, params):
        super().process(params)

        lookup_field = self.definition["lookup_field"]
        value_field = self.definition["value_field"]
        domain = [(lookup_field, "=", params)]

        record = self.env[self.definition["model"]].search_read(domain, [value_field])
        if len(record) > 0:
            res = record[0][value_field]
        else:
            res = None

        return res

File: nuido_flow_data/static/src/models/data/lookup_node.js
// THIS FILE IS A PART OF PUBLIC REPOSITORY https://github.com/yonitjio/exploring-odoo
// 
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT
// 
// THIS SOFTWARE IS EXPERIMENTAL AND FOR EDUCATIONAL PURPOSE ONLY.
// DO NOT USE IT IN PRODUCTION.
import { NodeModel } from "@nuido/models/node";
import { LookupPort } from "@nuido_flow_data/components/ports/lookup_port";
export class LookupNodeModel extends NodeModel {
    setup() {
        const auxOutId = "aux-out-" + this.id + "-1";
        this.addAuxOutPort(auxOutId, LookupPort.name, 1, {
            role: "lookup"
        });
        this.key = "my_lookup";
        this.model = "";
        this.model_description = "";
        this.lookup_field = "";
        this.value_field = "";
    }
}
